auth/realm/base/src/main/java/org/wildfly/security/auth/realm/FileSystemSecurityRealm.java

```diff
@@ -774,12 +830,26 @@ private Void replaceIdentityPrivileged(final LoadedIdentity newIdentity) throws
             }
         }
 
-        private void writeIdentity(final XMLStreamWriter streamWriter, final LoadedIdentity newIdentity) throws XMLStreamException, InvalidKeySpecException, NoSuchAlgorithmException, CertificateEncodingException {
+        private Version requiredVersion(final LoadedIdentity identityToWrite) {
+            // As new functionality is added we will identify if we need to use a later version
+            // if new functionality is used then use the required schema version otherwise fallback
+            // to an older version.
```

Comment:
If `secretKey` is not null, the required version should be 1.1.

Reply:
Changed